### PR TITLE
configure effects

### PR DESCRIPTION
**Pull request by jshvarts-sqsp**

### CODE REVIEW

File: app/src/main/java/com/jshvarts/todoapp/arch/MviViewModel.kt [hunk 1]
```diff
@@ -32,4 +23,8 @@ abstract class MviViewModel<A : UiAction, S : UiState, E : UiEffect>(
   fun dispatchAction(action: A) {
     return if (isStateInBundle) Unit else handleAction(action)
   }
+
+  interface HasUiEffect<E : UiEffect> {
```
Comment (anand-codes) [inline app/src/main/java/com/jshvarts/todoapp/arch/MviViewModel.kt:27]: How about `EffectSource`?

**Review by anand-codes — APPROVED** [review]
1 minor comment

File: app/src/main/java/com/jshvarts/todoapp/arch/MviViewModel.kt (hunk 1 above)
Comment (jshvarts-sqsp) [reply app/src/main/java/com/jshvarts/todoapp/arch/MviViewModel.kt:27]: how about `EffectProducer` or `ProducesEffect` or `EmitsEffects`?

Comment (jshvarts-sqsp) [reply app/src/main/java/com/jshvarts/todoapp/arch/MviViewModel.kt:27]: nit: to me `source` implies that making effects is it's primary function which is not

Comment (jshvarts-sqsp) [reply app/src/main/java/com/jshvarts/todoapp/arch/MviViewModel.kt:27]: will merge and rename in a follow up

Comment (anand-codes) [reply app/src/main/java/com/jshvarts/todoapp/arch/MviViewModel.kt:27]: I like `EffectProducer`

**Pull request merged**